src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1500,135 +1471,70 @@ public string[] GetDateTimeFormats(char format, IFormatProvider? provider)
         // IConvertible implementation
         //
 
-        public TypeCode GetTypeCode()
-        {
-            return TypeCode.DateTime;
-        }
-
-        bool IConvertible.ToBoolean(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Boolean"));
-        }
-
-        char IConvertible.ToChar(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Char"));
-        }
-
-        sbyte IConvertible.ToSByte(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "SByte"));
-        }
-
-        byte IConvertible.ToByte(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Byte"));
-        }
-
-        short IConvertible.ToInt16(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Int16"));
-        }
-
-        ushort IConvertible.ToUInt16(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "UInt16"));
-        }
-
-        int IConvertible.ToInt32(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Int32"));
-        }
-
-        uint IConvertible.ToUInt32(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "UInt32"));
-        }
-
-        long IConvertible.ToInt64(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Int64"));
-        }
-
-        ulong IConvertible.ToUInt64(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "UInt64"));
-        }
-
-        float IConvertible.ToSingle(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Single"));
-        }
+        public TypeCode GetTypeCode() => TypeCode.DateTime;
 
-        double IConvertible.ToDouble(IFormatProvider? provider)
-        {
-            throw new InvalidCastException(SR.Format(SR.InvalidCast_FromTo, "DateTime", "Double"));
-        }
+        bool IConvertible.ToBoolean(IFormatProvider? provider) => throw InvalidCast(nameof(Boolean));
```

Comment:
I do not think it is worth optimizing these using ThrowHelper.

Reply:
This isn't an optimization but just reducing code size/duplication.